h2/src/main/org/h2/value/DataType.java

```diff
@@ -1179,6 +1200,9 @@ public static boolean supportsAdd(int type) {
         case Value.SHORT:
             return true;
         default:
+            if (JdbcUtils.customDataTypesHandler != null) {
```

Comment:
This looks wrong: you should not ask customDataTypesHandler about builtin types.